lib/unleash/backup_file_reader.rb

```diff
@@ -0,0 +1,30 @@
+require 'unleash/configuration'
+require 'json'
+
+module Unleash
+  class BackupFileReader
+    def self.read!
+      Unleash.logger.debug "read!()"
+
+      backup_file = Unleash.configuration.backup_file
+      return nil unless File.exist?(backup_file)
+
+      File.read(backup_file)
+    rescue IOError => e
+      # :nocov:
+      Unleash.logger.error "Unable to read the backup_file: #{e}"
+      # :nocov:
+      nil
+    rescue JSON::ParserError => e
```

Comment:
I actually don't think this case can happen since we no longer parse on read. I realise this is a hangover from the old implementation (my bad, should have dropped this when I ported to Ygg). If we don't remove it in this PR, no biggie, I'll do it in a future one